substratest/assets.py

```diff
@@ -396,28 +408,21 @@ class Meta:
         }
 
 
-@dataclasses.dataclass
-class ComputePlan(_Asset, _ComputePlanFutureMixin):
+class ComputePlan(_ComputePlanFutureAsset):
     compute_plan_id: str
     status: str
-    traintuple_keys: typing.List[str]
-    composite_traintuple_keys: typing.List[str]
-    aggregatetuple_keys: typing.List[str]
-    testtuple_keys: typing.List[str]
+    traintuple_keys: typing.List[str] = None
+    composite_traintuple_keys: typing.List[str] = None
+    aggregatetuple_keys: typing.List[str] = None
+    testtuple_keys: typing.List[str] = None
     tag: str
 
-    def __post_init__(self):
-        if self.traintuple_keys is None:
-            self.traintuple_keys = []
-
-        if self.composite_traintuple_keys is None:
-            self.composite_traintuple_keys = []
-
-        if self.aggregatetuple_keys is None:
-            self.aggregatetuple_keys = []
-
-        if self.testtuple_keys is None:
-            self.testtuple_keys = []
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.traintuple_keys = self.traintuple_keys or []
+        self.composite_traintuple_keys = self.composite_traintuple_keys or []
+        self.aggregatetuple_keys = self.aggregatetuple_keys or []
+        self.testtuple_keys = self.testtuple_keys or []
```

Comment:
I think we could remove the `= None` by simply inversing the order of things in the `__init__`:

```python
def __init__(self, *args, **kwargs):
    kwargs['traintuple_keys'] = kwargs.get('traintuple_keys') or []
    kwargs['composite_traintuple_keys'] = kwargs.get('composite_traintuple_keys') or []
    kwargs['aggregatetuple_keys'] = kwargs.get('aggregatetuple_keys') or []
    kwargs['testtuple_keys'] = kwargs.get('testtuple_keys') or []
    super().__init__(*args, **kwargs)
```

Reply:
> I just discovered that pydantic allows to use a `@dataclass` decorator similar to the one from dataclasses: https://pydantic-docs.helpmanual.io/usage/dataclasses/ It looks like it'd be a bit simpler than these new classes inheriting from pydantic.basemodel. What was the benefit of using BaseModel over this decorator?

Using a dataclass decorator or inheriting from a  schema class are both good solutions to me. What would be interesting to know is, do the modifications of classes inheritance are required (removing Mixin classes and multiple inheritance)?